File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/C++/s684030429_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Main {
private:
    int max = 0, cur = 0;
    int ni() {
        int a;
        cin >> a;
        return a;
    }
    int nl() {
        long a;
        cin >> a;
        return a;
    }
    float nf() {
        float a;
        cin >> a;
        return a;
    }
    double nd() {
        double a;
        cin >> a;
        return a;
    }
    string ns() {
        string s;
        getline(cin, s);
        return s;
    }
    void ia(vector<int>& a) {
        for (int i = 0; i < a.size(); i++) {
            a[i] = ni();
        }
    }
    void la(vector<long>& a) {
        for (int i = 0; i < a.size(); i++) {
            a[i] = nl();
        }
    }
    void fa(vector<float>& a) {
        for (int i = 0; i < a.size(); i++) {
            a[i] = nf();
        }
    }
    void da(vector<double>& a) {
        for (int i = 0; i < a.size(); i++) {
            a[i] = nd();
        }
    }
    void sa(vector<string>& a, bool empty) {
        if (empty) {
            for (int i = 0; i < a.size(); i++) {
                a[i] = "";
            }
        } else {
            for (int i = 0; i < a.size(); i++) {
                a[i] = ns();
            }
        }
    }
    void ida(vector<vector<int>>& a, int n, int m) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                a[i][j] = ni();
            }
        }
    }
    void lda(vector<vector<long>>& a, int n, int m) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                a[i][j] = nl();
            }
        }
    }
    void dda(vector<vector<double>>& a, int n, int m) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                a[i][j] = nd();
            }
        }
    }
    int stoi(string s) {
        return stoi(s);
    }
    double stod(string s) {
        return stod(s);
    }
    long lmin(vector<long>& a) {
        long min = LONG_MAX;
        for (int i = 0; i < a.size(); i++) {
            if (min > a[i]) min = a[i];
        }
        return min;
    }
    int imin(vector<int>& a) {
        int min = INT_MAX;
        for (int i = 0; i < a.size(); i++) {
            if (min > a[i]) min = a[i];
        }
        return min;
    }
    long lmax(vector<long>& a) {
        long max = LONG_MIN;
        for (int i = 0; i < a.size(); i++) {
            if (max < a[i]) max = a[i];
        }
        return max;
    }
    int imax(vector<int>& a) {
        int max = INT_MIN;
        for (int i = 0; i < a.size(); i++) {
            if (max < a[i]) max = a[i];
        }
        return max;
    }
    bool ibs(vector<int>& a, int toFind) {
        sort(a.begin(), a.end());
        int min = 0;
        int max = a.size() - 1;
        bool found = false;

        while (min <= max && !found) {
            int mid = min + (max - min) / 2;
            if (a[mid] == toFind) {
                found = true;
            } else if (toFind > a[mid]) {
                min = mid + 1;
            } else {
                max = mid - 1;
            }
        }
        return found;
    }
    bool lbs(vector<long>& a, long toFind) {
        sort(a.begin(), a.end());
        int min = 0;
        int max = a.size() - 1;
        bool found = false;

        while (min <= max && !found) {
            int mid = min + (max - min) / 2;
            if (a[mid] == toFind) {
                found = true;
            } else if (toFind > a[mid]) {
                min = mid + 1;
            } else {
                max = mid - 1;
            }
        }
        return found;
    }
    int stb(string s) {
        int sum = 0;
        int k = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            sum += stoi(s.substr(i, 1)) * pow(2, k++);
        }
        return sum;
    }
public:
    void solve() {
        int a = ni();
        int b = ni();

        cout << max(0, a - 2 * b) << endl;
    }
};

int main() {
    Main().solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s684030429_commented.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

type Main struct {
	sc *Scanner
	max int
	cur int
}

func NewMain() *Main {
	return &Main{sc: new(Scanner)}
}

func (m *Main) solve() {
	a := m.ni()
	b := m.ni()

	fmt.Println(math.Max(0, float64(a-2*b)))
}

func (m *Main) ni() int {
	var a int
	fmt.Scan(&a)
	return a
}

func (m *Main) nl() int64 {
	var a int64
	fmt.Scan(&a)
	return a
}

func (m *Main) nf() float32 {
	var a float32
	fmt.Scan(&a)
	return a
}

func (m *Main) nd() float64 {
	var a float64
	fmt.Scan(&a)
	return a
}

func (m *Main) ns() string {
	var s string
	fmt.Scan(&s)
	return s
}

func (m *Main) ia(a []int) {
	for i := range a {
		a[i] = m.ni()
	}
}

func (m *Main) la(a []int64) {
	for i := range a {
		a[i] = m.nl()
	}
}

func (m *Main) fa(a []float32) {
	for i := range a {
		a[i] = m.nf()
	}
}

func (m *Main) da(a []float64) {
	for i := range a {
		a[i] = m.nd()
	}
}

func (m *Main) sa(a []string, empty bool) {
	if empty {
		for i := range a {
			a[i] = ""
		}
	} else {
		for i := range a {
			a[i] = m.ns()
		}
	}
}

func (m *Main) ida(a [][]int, n, m int) {
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			a[i][j] = m.ni()
		}
	}
}

func (m *Main) lda(a [][]int64, n, m int) {
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			a[i][j] = m.nl()
		}
	}
}

func (m *Main) dda(a [][]float64, n, m int) {
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			a[i][j] = m.nd()
		}
	}
}

func (m *Main) stoi(s string) int {
	a, _ := strconv.Atoi(s)
	return a
}

func (m *Main) stod(s string) float64 {
	a, _ := strconv.ParseFloat(s, 64)
	return a
}

func (m *Main) lmin(a []int64) int64 {
	min := int64(math.MaxInt64)
	for _, v := range a {
		if v < min {
			min = v
		}
	}
	return min
}

func (m *Main) imin(a []int) int {
	min := math.MaxInt32
	for _, v := range a {
		if v < min {
			min = v
		}
	}
	return min
}

func (m *Main) lmax(a []int64) int64 {
	max := int64(math.MinInt64)
	for _, v := range a {
		if v > max {
			max = v
		}
	}
	return max
}

func (m *Main) imax(a []int) int {
	max := math.MinInt32
	for _, v := range a {
		if v > max {
			max = v
		}
	}
	return max
}

func (m *Main) ibs(a []int, toFind int) bool {
	sort.Ints(a)
	min := 0
	max := len(a) - 1
	found := false

	for min <= max && !found {
		mid := min + (max-min)/2
		if a[mid] == toFind {
			found = true
		} else if toFind > a[mid] {
			min = mid + 1
		} else {
			max = mid - 1
		}
	}
	return found
}

func (m *Main) lbs(a []int64, toFind int64) bool {
	sort.Slice(a, func(i, j int) bool { return a[i] < a[j] })
	min := 0
	max := len(a) - 1
	found := false

	for min <= max && !found {
		mid := min + (max-min)/2
		if a[mid] == toFind {
			found = true
		} else if toFind > a[mid] {
			min = mid + 1
		} else {
			max = mid - 1
		}
	}
	return found
}

func (m *Main) stb(s string) int {
	sum := 0
	k := 0
	for i := len(s) - 1; i >= 0; i-- {
		sum += m.stoi(string(s[i])) * int(math.Pow(2, float64(k)))
		k++
	}
	return sum
}

func main() {
	mainObj := NewMain()
	mainObj.solve()
}
// End of Code
